osbs/utils.py

```diff
@@ -316,12 +318,14 @@ def make_name_from_git(repo, branch, limit=53, separator='-'):
 
     repo = git_repo_humanish_part_from_uri(repo)
     branch = branch or 'unknown'
+    hash_str = b64encode(sha256(repo + branch).hexdigest())[:hash_size]
```

Comment:
So actually this should be more like:
```
full = repo + branch
hash_str = urlsafe_b64encode(sha256(full.encode('utf-8')).digest())[:hash_size]
if hash_str.endswith('-'):
    # Replace final '-' with '_' to make a valid name
    hash_str = hash_str[:len(hash_str) - 1] + '_'
```

urlsafe_b64encode() avoids '/' and '+'.